### PR TITLE
Products - empty view pull to refresh

### LINKED ISSUE #6356: Cannot refresh lists when an empty view is present
# Description
For most of the lists I've found in the app: Orders, Products, Reviews, Plugins, Coupons - when an empty view is present, it's not possible to pull down the view to trigger refreshing data. For Orders and Products, this is more problematic because there's no other way to fetch data other than killing and relaunching the app.

# Steps to reproduce
- Login to the app with a valid account
- Turn off the internet connection
- Switch to the Orders or Products tab. Without internet connection, no data can be fetched, so empty views will be presented shortly afterward.
- Turn on the internet connection. Notice that there's no way to reload order and product lists. 

# Expectation
We should be able to pull down to refresh the lists even with the empty views in place.

### PR DESCRIPTION
**Pull request by selanthiraiyan**
Closes: #6356

### Description
This PR adds pull to refresh to the empty view controller in `ProductsViewController`.
- Add a pull to refresh callback to empty view when there are no products available.
- Stop adding `EmptyStateViewController` more than once. i.e. Abort if empty view is already presented. (This fix is already present in `OrderListViewController`, `ReviewsViewController`.)

### Testing instructions

- Login to the app with a valid account
- Turn off the internet connection
- Switch to the Products tab. Without internet connection, no data can be fetched, so empty views will be presented shortly afterward.
- Turn on the internet connection.
- Pull down to refresh and observe that products are loaded now.

### Screenshots

https://user-images.githubusercontent.com/524475/171548790-35cba363-c405-4191-bb5d-18d4d5598c7c.MP4

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/ViewRelated/Products/ProductsViewController.swift [hunk 1]
```diff
@@ -756,9 +760,13 @@ private extension ProductsViewController {
...
-            buttonTitle: buttonTitle) { [weak self] button in
-            self?.addProduct(sourceView: button)
-        }
+            buttonTitle: buttonTitle,
+            onTap: { [weak self] button in
+                self?.addProduct(sourceView: button)
+            },
+            onPullToRefresh: { [weak self] refreshControl in
+                self?.pullToRefresh(sender: refreshControl)
+            })
```
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Products/ProductsViewController.swift:769]: nit: similar to the orders PR, can we also add this to the empty results view from filters in `createNoProductsMatchFilterConfig`?

**Review by jaclync — APPROVED** [review]
Thanks for improving the empty state UX! LGTM 🚢 

Unrelated to this PR, the animation after pull-to-refresh feels a bit jarring when the results stay empty (refreshing when the device remains offline) where the offline banner appears for a split second:


https://user-images.githubusercontent.com/1945542/171566070-cd1080fd-1949-4d92-b9ad-79d7631f7964.MP4

did you also see this?

File: WooCommerce/Classes/ViewRelated/Products/ProductsViewController.swift (hunk 1 above)
Comment (selanthiraiyan) [reply WooCommerce/Classes/ViewRelated/Products/ProductsViewController.swift:769]: Added in https://github.com/woocommerce/woocommerce-ios/pull/7030/commits/a0061ed3a8a1722d2bef087094fc4a8e487c8331

**Comment by selanthiraiyan** [comment]
> Unrelated to this PR, the animation after pull-to-refresh feels a bit jarring when the results stay empty (refreshing when the device remains offline) where the offline banner appears for a split second:

Great catch. 👏 

I couldn't notice the offline banner appearing. But, I have logged a issue for the animation [here](https://github.com/woocommerce/woocommerce-ios/issues/7037).

**Pull request merged**